homeassistant/components/bluecurrent/__init__.py

```diff
@@ -0,0 +1,244 @@
+"""The BlueCurrent integration."""
+from __future__ import annotations
+
+from datetime import datetime
+from typing import Any
+
+from bluecurrent_api import Client
+from bluecurrent_api.exceptions import (
+    BlueCurrentException,
+    InvalidApiToken,
+    RequestLimitReached,
+    WebsocketException,
+)
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_API_TOKEN, EVENT_HOMEASSISTANT_STOP
+from homeassistant.core import Event, HomeAssistant, callback
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryNotReady
+from homeassistant.helpers import entity_registry
+from homeassistant.helpers.dispatcher import (
+    async_dispatcher_connect,
+    async_dispatcher_send,
+)
+from homeassistant.helpers.entity import DeviceInfo, Entity
+from homeassistant.helpers.event import async_call_later
+
+from .const import (
+    CHARGE_POINTS,
+    DATA,
+    DELAY_1,
+    DELAY_2,
+    DOMAIN,
+    EVSE_ID,
+    GRID_STATUS,
+    LOGGER,
+    MODEL_TYPE,
+    OBJECT,
+    PLATFORMS,
+    VALUE_TYPES,
+)
+
+
+async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Set up Blue Current as a config entry."""
+    hass.data.setdefault(DOMAIN, {})
+    client = Client()
+    api_token = config_entry.data[CONF_API_TOKEN]
+    connector = Connector(hass, config_entry, client)
+    try:
+        await connector.connect(api_token)
+    except (BlueCurrentException) as err:
+        if isinstance(err, InvalidApiToken):
+            raise ConfigEntryAuthFailed("Invalid api token.") from err
+        raise ConfigEntryNotReady from err
```

Comment:
You can have Python do the `isinstance` with
```suggestion
    except InvalidApiToken as err:
        raise ConfigEntryAuthFailed("Invalid API token.") from err
    except BlueCurrentException as err:
        raise ConfigEntryNotReady from err
```

Reply:
Changed.